search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/prompt/PromptUtil.java

```diff
@@ -75,25 +89,56 @@ enum ChatRole {
     }
 
     @VisibleForTesting
-    static String buildMessageParameter(String question, List<Interaction> chatHistory, List<String> contexts) {
+    static String buildMessageParameter(String systemPrompt, String userInstructions, String question, List<Interaction> chatHistory, List<String> contexts) {
 
         // TODO better prompt template management is needed here.
 
+        if (Strings.isNullOrEmpty(systemPrompt) && Strings.isNullOrEmpty(userInstructions)) {
+            systemPrompt = DEFAULT_SYSTEM_PROMPT;
+        }
```

Comment:
If only userInstructions exists, we don't set default system promt for chat use case?

Reply:
Yes, that's right.  Actually, you can pass a system prompt as part of user instructions.  We are getting some mixed results and keep the behavior this way so we can do more experiments.